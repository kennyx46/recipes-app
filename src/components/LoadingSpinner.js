import React from 'react';
import Container from 'react-bootstrap/Container';
import Spinner from 'react-bootstrap/Spinner';

const LoadingSpinner = () => {
	return (
		<Container className="text-center mt-4">
		    <Spinner size="lg" animation="border"/>
		</Container>
	);
};

export default LoadingSpinner;
