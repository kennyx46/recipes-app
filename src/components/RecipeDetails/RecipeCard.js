import React from 'react';
import ReactMarkdown from 'react-markdown';

import Card from 'react-bootstrap/Card';
import Badge from 'react-bootstrap/Badge';

const RecipeCard = ({ image, title, tags, chef, description }) => {
	return (
		<Card>
		    <Card.Img variant="top" src={image} />
		    <Card.Body>
		        <Card.Title>{title}</Card.Title>
		        <div>
		            {tags.map((tag) => <Badge key={tag} className="mr-2 mb-3" variant="secondary">{tag}</Badge>)}
		        </div>
		      	<ReactMarkdown source={description} />
		    </Card.Body>
		    {chef && <Card.Footer className="font-weight-bold">Chef: {chef}</Card.Footer>}
		</Card>
	);
}

export default RecipeCard;
