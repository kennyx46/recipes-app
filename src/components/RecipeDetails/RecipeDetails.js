import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Alert from 'react-bootstrap/Alert';
import LoadingSpinner from '../LoadingSpinner';
import RecipeCard from './RecipeCard';

export default class RecipeDetails extends Component {

    componentDidMount() {
        const { recipeId } = this.props.match.params;

        this.props.getRecipe({ recipeId });
    }

    render() {
        const { isLoading, recipe, error } = this.props;

        if (isLoading) {
            return (
                <LoadingSpinner/>
            );
        }

        return (
            <Container>
                <h1>Recipe Details</h1>
                <div className="my-2">
                    <Link to='/'>Recipes</Link>
                </div>
                { error || !recipe ?
                    <Alert variant='danger'>Someting went wrong. Please try again later</Alert> :
                    <RecipeCard {...recipe}/>
                }
            </Container>
        );
    }
}
