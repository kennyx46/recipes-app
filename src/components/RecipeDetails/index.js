import { connect } from 'react-redux';
import { compose } from 'redux';
import { getRecipe, } from '../../data/recipes/actions';
import RecipeDetails from './RecipeDetails';
import { getRecipesIsLoading, getRecipesError, getRecipeItem } from '../../data/recipes/selectors';
import { withRouter } from "react-router";

const mapStateToProps = (state) => ({
    recipe: getRecipeItem(state),
    isLoading: getRecipesIsLoading(state),
    error: getRecipesError(state),
});

const mapDispatchToProps = {
    getRecipe,
};

export default compose(
    connect(mapStateToProps, mapDispatchToProps),
    withRouter,
)(RecipeDetails);
