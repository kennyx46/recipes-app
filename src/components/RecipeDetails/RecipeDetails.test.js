import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from './RecipeCard';

import RecipeDetails from './RecipeDetails';
import LoadingSpinner from '../LoadingSpinner';

describe('<RecipeDetails>', () => {
    const defaultProps = Object.freeze({
        recipe: {
            tags: []
        },
        isLoading: false,
        error: null,
        match: { params: {} },
        getRecipe: jest.fn(),
    });

    it('renders component', async () => {
        const wrapper = shallow(
            <RecipeDetails {...defaultProps }/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('shows <LoadingSpinner> if loading in progress', async () => {
        const props = { ...defaultProps, isLoading: true };
        const recipeDetails = shallow(<RecipeDetails {...props} />);

        expect(recipeDetails.exists(LoadingSpinner)).toBeTruthy();
    });

    it('renders <RecipeCard> if recipe is present', async () => {
        const props = { ...defaultProps };
        const recipeDetails = shallow(<RecipeDetails {...props} />);

        expect(recipeDetails.find(RecipeCard)).toHaveLength(1);
    });

});
