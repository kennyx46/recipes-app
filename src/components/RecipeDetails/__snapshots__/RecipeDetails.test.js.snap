// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetails> renders component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeDetails
    error={null}
    getRecipe={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "recipeId": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    isLoading={false}
    match={
      Object {
        "params": Object {},
      }
    }
    recipe={
      Object {
        "tags": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1>
          Recipe Details
        </h1>,
        <div
          className="my-2"
        >
          <Link
            replace={false}
            to="/"
          >
            Recipes
          </Link>
        </div>,
        <RecipeCard
          tags={Array []}
        />,
      ],
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Recipe Details",
        },
        "ref": null,
        "rendered": "Recipe Details",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
          >
            Recipes
          </Link>,
          "className": "my-2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Recipes",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Recipes",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "tags": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1>
            Recipe Details
          </h1>,
          <div
            className="my-2"
          >
            <Link
              replace={false}
              to="/"
            >
              Recipes
            </Link>
          </div>,
          <RecipeCard
            tags={Array []}
          />,
        ],
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Recipe Details",
          },
          "ref": null,
          "rendered": "Recipe Details",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              Recipes
            </Link>,
            "className": "my-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Recipes",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Recipes",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "tags": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
